prince-of-versions/src/test/java/co/infinum/princeofversions/tests/JSONParserTest.java

```diff
@@ -61,6 +62,105 @@ public void testParsingContentJSONWhenCurrentIsLessThanMinAndLessThanOptional()
         }
     }
 
+
+    @Test
+    public void testParsingContentJSONWhenCurrentIsLessThanMinAndLessThanOptionalBetaBeta() {
+        VersionContext.Version currentVersion = new VersionContext.Version("1.2.3-beta1");
+        try {
+            JsonVersionConfigParser parser = new JsonVersionConfigParser(currentVersion);
+            JSONObject parsedResponseContent = new JSONObject(ResourceUtils.readFromFile("valid_update_full_beta.json"));
+            VersionContext version = parser.parse(parsedResponseContent);
+            assertEquals("Current version should be 1.0.0", "1.2.3-beta1", version.getCurrentVersion().getVersionString());
```

Comment:
Copy paste is a bad thing, try to avoid it. In most of the examples you have "Current version should be 1.0.0", and the current version is never 1.0.0. The same happens with the optional version in several occasions. Besides that, tests are great.